Assets/[ProjectRei]/Scripts/Runtime/Character/Character.cs

```diff
@@ -0,0 +1,97 @@
+using System;
+using UnityEngine;
+using CombatSystem;
+
+namespace CharacterSystem
+{
+    [DisallowMultipleComponent]
+    public sealed class Character : MonoBehaviour, ICombatant
+    {
+        #region Fields
+        [SerializeField, Min(1)]
+        private int m_hitPoints = 100;
+
+        [Space]
+
+        [SerializeField]
+        private CharacterInput m_input = null;
+
+        [SerializeField]
+        private CharacterLocomotor m_locomotor = null;
+
+        private int m_currentHitPoints = 0;
+        
+        #endregion
+
+
+        #region Events
+        public event Action Revived;
+        public event Action Died;
+        public event Action<int> Damaged;
+        public event Action<int> Healed;
+        public event HitPointEventHandler HitPointsUpdated;
+        #endregion
+
+
+        #region Properties
+        public int currentHP => m_currentHitPoints;
+        public int totalHP => m_hitPoints;
+        public bool isAlive => (m_currentHitPoints > 0);
+        #endregion
+
+
+        #region MonoBehaviour Implementation
+        private void Start() => ResetCurrentHP();
+        private void Update() => UpdateLocomotion();
+        #endregion
+
+
+        #region Public Methods
+        public void ReduceHitPoints(int value)
+        {
+            if (!isAlive)
+                return;
+
+            value = Mathf.Max(value, 0);
+            m_currentHitPoints -= value;
+            Damaged?.Invoke(value);
+            HitPointsUpdated?.Invoke(m_currentHitPoints, m_hitPoints);
+
+            if (!isAlive)
+                Died?.Invoke();
+        }
+
+        public void RestoreHitPoints(int value)
+        {
+            if (!isAlive || m_currentHitPoints >= m_hitPoints)
+                return;
+
+            value = Mathf.Max(value, 0);
+            m_currentHitPoints = Mathf.Min(m_currentHitPoints + value, m_hitPoints);
+            Healed?.Invoke(value);
+            HitPointsUpdated?.Invoke(m_currentHitPoints, m_hitPoints);
+        }
+        #endregion
+
+
+        #region Internal Methods
+        private void ResetCurrentHP()
+        {
+            bool wasDead = !isAlive;
+            m_currentHitPoints = m_hitPoints;
+
+            if (wasDead)
```

Comment:
Minor nitpick. Do we need to assign !isAlive to wasDead variable? I only see this as for added readability